src/Nethermind/Nethermind.Synchronization/SyncServer.cs

```diff
@@ -249,34 +254,70 @@ private void SyncBlock(Block block, ISyncPeer? syncPeer)
             {
                 if (!_blockValidator.ValidateSuggestedBlock(block))
                 {
-                    string message = $"Peer {syncPeer?.Node:c} sent an invalid block.";
+                    string message = $"Peer {nodeWhoSentTheBlock?.Node:c} sent an invalid block.";
                     if (_logger.IsDebug) _logger.Debug(message);
-                    lock (_recentlySuggested)
-                    {
-                        _recentlySuggested.Delete(block.Hash!);
-                    }
-
+                    _recentlySuggested.Delete(block.Hash!);
```

Comment:
oh maybe lock is no longer needed, but why?

Reply:
LruKeyCache is thread-safe by design